packages/apollo-language-server/src/diagnostics.ts

```diff
@@ -11,25 +13,43 @@ import { GraphQLDocument } from "./document";
 import { highlightNodeForNode } from "./utilities/graphql";
 import { rangeForASTNode } from "./utilities/source";
 
+import { TypeKind } from "apollo/lib/printer/ast";
 import { getValidationErrors } from "apollo/lib/validation";
 
-export function collectDiagnostics(
+/**
+ * Build an array of code diagnostics for all GraphQL operations in a document.
+ * This function explicitly ignores type definitions.
+ */
+export function collectOperationDiagnostics(
   schema: GraphQLSchema,
   queryDocument: GraphQLDocument,
   fragments: { [fragmentName: string]: FragmentDefinitionNode }
 ): Diagnostic[] {
   const ast = queryDocument.ast;
   if (!ast) return queryDocument.syntaxErrors;
 
+  // I would prefer to use the TypeKind enum from graphql-js here,
+  // but the enum values aren't the same, i.e. "OBJECT" vs "ObjectTypeDefinition"
+  const typeKinds = Object.values(TypeKind);
```

Comment:
You should be able to import `Kind` from `graphql` and check against those values. But you probably want to use utility methods like `isTypeSystemDefinitionNode` and `isTypeSystemExtensionNode` instead, so you don't have to enumerate all possible values.

Reply:
Ah, it turns out there is also `isExecutableDefinitionNode`, which checks for operation and fragment definitions.